includes/class-access-token.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * File containing the class Access_Token.
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * A token that provides access to a private resource.
+ *
+ * @since $$next-version$$
+ */
+class Access_Token {
+
+	/**
+	 * An expiry timestamp.
+	 *
+	 * @var int|null
+	 */
+	private ?int $expiry;
+
+	/**
+	 * Token's payload. It is hashed together with the token.
+	 *
+	 * @var array
+	 */
+	private array $payload;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array    $payload A payload to be hashed together with the token.
+	 * @param int|null $expiry Expiry timestamp.
+	 */
+	public function __construct( array $payload, int $expiry = null ) {
```

Comment:
So if I understand correctly, the secret URL would look like `/alerts?user_id=1232&token=XXXXXXXX`? Could we encode the user ID in the token?

Reply:
Sorry the question is, do we need to include the `user_id` as a parameter too in the link?

Reply:
Yes. So if we want the user id to be included in the token, then we will need to use a symmetric key cipher which I think adds extra complexity without gaining much.